glusterd2/commands/snapshot/snapshot-delete.go

```diff
@@ -0,0 +1,167 @@
+package snapshotcommands
+
+import (
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/snapshot"
+	"github.com/gluster/glusterd2/glusterd2/snapshot/lvm"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func snapshotBrickDelete(errCh chan error, wg *sync.WaitGroup, snapVol volume.Volinfo, b brick.Brickinfo, logger log.FieldLogger) {
+	defer wg.Done()
+
+	if snapVol.State == volume.VolStarted {
+		if err := snapshot.StopBrick(b, logger); err != nil {
+			log.WithError(err).WithField(
+				"brick", b.Path).Warning("Failed to cleanup the brick.Earlier it might have stopped abnormaly")
+
+		}
+	}
+	if err := lvm.RemoveBrickSnapshot(b.MountInfo.DevicePath); err != nil {
+		log.WithError(err).WithField(
+			"brick", b.Path).Debug("Failed to remove snapshotted LVM")
+		errCh <- err
+		return
+	}
+	mountRoot := strings.TrimSuffix(b.Path, b.MountInfo.Mountdir)
+	os.RemoveAll(mountRoot)
+	errCh <- nil
+	return
+}
+
+func snapshotDelete(c transaction.TxnCtx) error {
+	var snapinfo snapshot.Snapinfo
+	if err := c.Get("snapinfo", &snapinfo); err != nil {
+		return err
+	}
+
+	snapVol := snapinfo.SnapVolinfo
+	var wg sync.WaitGroup
+	numBricks := len(snapVol.GetBricks())
+	errCh := make(chan error, numBricks)
+
+	for _, b := range snapVol.GetLocalBricks() {
+		wg.Add(1)
+		go snapshotBrickDelete(errCh, &wg, snapVol, b, c.Logger())
+	}
+	go func() {
+		wg.Wait()
+		close(errCh)
+	}()
+	//var err error
+	err := error(nil)
+	for i := range errCh {
+		if i != nil && err == nil {
+			//Return the first error from goroutines
```

Comment:
Not returned any error

Reply:
done

Reply:
This err will be returned from the main return, we cannot just add a return as there are goroutines running.